templates/single-tapestry.php

```diff
@@ -41,17 +43,28 @@ function addNonceToScript()
                 comments_template();
             endif;
             ?>
+        
+        <?php endwhile; // end of the loop. ?>
 
-        <?php endwhile;
-    ?>
+        <div id="tapestry">
+            <!-- Don't render add root button when there's nodes -->
+            <?php if(!doesTapestryHaveNodes(get_the_ID())) : ?>
+                <div id="root-node-container">
+                    <div id="root-node-btn"><i class="fas fa-plus-circle fa-5x"></i></div>
+                    <div id="root-node-label">Add Root Node</div>
+                <div>
+            <?php endif; ?>
+        </div>
+        
+        <?php require "modal/create-new-node-modal.php";?>
```

Comment:
Could you please pass the full URL to this template into tapestry.js and then load it in tapestry.js via an ajax call using the URL? This way, you don't have to include it here. You could also consider changing it to be an html file instead.

Reply:
done.